crates/storage/db/src/abstraction/cursor.rs

```diff
@@ -122,6 +122,32 @@ impl<'cursor, 'tx, T: Table, CURSOR: DbCursorRO<'tx, T>> std::iter::Iterator
     }
 }
 
+/// Provides a reverse iterator to `Cursor` when handling `Table`.
+/// Also check [`Walker`]
+pub struct RWalker<'cursor, 'tx, T: Table, CURSOR: DbCursorRO<'tx, T>> {
+    /// Cursor to be used to walk through the table.
+    pub cursor: &'cursor mut CURSOR,
+    /// `(key, value)` where to start the walk.
+    pub start: IterPairResult<T>,
+    /// Phantom data for 'tx. As it is only used for `DbCursorRO`.
+    pub _tx_phantom: PhantomData<&'tx T>,
+}
+
+impl<'cursor, 'tx, T: Table, CURSOR: DbCursorRO<'tx, T>> std::iter::Iterator
```

Comment:
Let's add a method called rev() on walker which converts it to a ReverseWalker and vice versa?

Reply:
like the idea^